src/omv/modules/py_tf.c

```diff
@@ -774,6 +728,107 @@ STATIC mp_obj_t py_tf_detect(uint n_args, const mp_obj_t *pos_args, mp_map_t *kw
+                                                params->input_scale) + params->input_zero_point);
+            }
+        }
+    } else if (len == params->input_height) {
+        for (size_t i = 0; i < len; i++) {
+            size_t row_len;
+            mp_obj_t *row_items;
+            mp_obj_get_array(items[i], &row_len, &row_items);
+
+            if (row_len == (params->input_width * params->input_channels)) {
+                if (params->input_datatype == LIBTF_DATATYPE_FLOAT) {
+                    float *model_input_float = (float *) model_input;
+                    for (size_t j = 0; j < row_len; j++) {
+                        size_t index = (i * row_len) + j;
+                        model_input_float[index] = mp_obj_get_float(row_items[index]);
+                    }
+                } else {
+                    uint8_t *model_input_8 = (uint8_t *) model_input;
+                    for (size_t j = 0; j < row_len; j++) {
+                        size_t index = (i * row_len) + j;
+                        model_input_8[index] = fast_roundf((mp_obj_get_float(row_items[index]) /
+                                                            params->input_scale) + params->input_zero_point);
+                    }
+                }
+            } else if (row_len == params->input_height) {
+                for (size_t j = 0; j < row_len; j++) {
+                    size_t c_len;
+                    mp_obj_t *c_items;
+                    mp_obj_get_array(row_items[i], &c_len, &c_items);
+
+                    if (c_len == params->input_channels) {
+                        if (params->input_datatype == LIBTF_DATATYPE_FLOAT) {
+                            float *model_input_float = (float *) model_input;
+                            for (size_t k = 0; k < c_len; k++) {
+                                size_t index = (i * row_len) + (j * c_len) + k;
+                                model_input_float[index] = mp_obj_get_float(c_items[index]);
+                            }
+                        } else {
+                            uint8_t *model_input_8 = (uint8_t *) model_input;
+                            for (size_t k = 0; k < c_len; k++) {
+                                size_t index = (i * row_len) + (j * c_len) + k;
+                                model_input_8[index] = fast_roundf((mp_obj_get_float(c_items[index]) /
+                                                                    params->input_scale) + params->input_zero_point);
+                            }
+                        }
+                    } else {
+                        mp_raise_msg(&mp_type_ValueError, MP_ERROR_TEXT("Channel count mismatch!"));
+                    }
+                }
+            } else {
+                mp_raise_msg(&mp_type_ValueError, MP_ERROR_TEXT("Column count mismatch!"));
+            }
+        }
+    } else {
+        mp_raise_msg(&mp_type_ValueError, MP_ERROR_TEXT("Row count mismatch!"));
+    }
```

Comment:
Can you do these sanity checks early in the function so it's more readable ?

Reply:
I would have to duplicate the same if statements and logic to do this. Are you sure? For the nested raises I don't have the information I need until I call mp_obj_get_array.

Reply:
Actually why not just support a flat array ? One can easily flatten the array in Python with ulab ndarray.flatten, or pure Python:
```python
[j for sub in [[1,2,3],[4,5,6],[7,8,9]] for j in sub]
```

Reply:
Seems like this would be less user-friendly. I've noticed a few different forum threads and folks opening up issue posts about this function such that it's worth it to make it handle any array people could throw at it. Right now, it will handle a 1d array, a 2d array, and a 3d array without issue.

Reply:
Yeah but it's more text, code adds up and then we run out of flash. I'll merge it though.